lib/src/platform_text_field.dart

```diff
@@ -387,6 +438,8 @@ class PlatformTextField
   TextField createMaterialWidget(BuildContext context) {
     final data = material?.call(context, platform(context));
 
+    final decoration = _inputDecorationWithHint(hint, data?.decoration ?? const InputDecoration());
```

Comment:
```suggestion
    final decoration = hint == null?
         data?.decoration ?? const InputDecoration()
         _inputDecorationWithHint(hint, data?.decoration)
```

No need to recreate the InputDecoration if there is no hint.